pentasharp/Services/TaxiBookingService.cs

```diff
@@ -101,5 +101,39 @@ public async Task<bool> UpdateBookingAsync(EditTaxiBookingViewModel model)
             await _context.SaveChangesAsync();
             return true;
         }
+
+        public async Task<List<TaxiBookingViewModel>> GetBookingsForUserAsync(int userId)
+        {
+            var bookings = await _context.TaxiBookings
+                   .Include(r => r.User)
+                     .Where(r => r.UserId == userId)
+                     .Include(r => r.Taxi)
+                     .Include(r => r.TaxiCompany)
+                     .Where(r => r.UserId == userId)
+                .ToListAsync();
+
+            return _mapper.Map<List<TaxiBookingViewModel>>(bookings);
+        }
+
+        public async Task<bool> CancelBookingAsync(int bookingId, int userId)
+        {
+            var booking = await _context.TaxiBookings
```

Comment:
you can improve this by adding a check for status here, it would be 
 var booking = await _context.TaxiBookings.FirstOrDefaultAsync(b => b.BookingId == bookingId && b.UserId == userId && b.Status == ReservationStatus.Pending);
 
 in that note this if condition             if (booking.Status != ReservationStatus.Pending)
            {
                return false; // Only bookings with a "Pending" status can be canceled
            }
can be ereased. Also remove explanatory comments